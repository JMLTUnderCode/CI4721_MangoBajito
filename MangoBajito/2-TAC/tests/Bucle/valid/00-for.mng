# Test 00: Bucle FOR.
se_prende(){
	culito n: mango = 5;
	repite_burda i entre 1 hasta 4 {
		n = n + i;
	};
};
