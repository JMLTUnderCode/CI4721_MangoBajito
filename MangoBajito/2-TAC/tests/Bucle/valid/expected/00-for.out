| AcomÃ³date que te lo estan analizando d:-)
--> Print table? (s/n): [1;36m[5m
               =======================================================
                             Symbol Table Representation              
               =======================================================
[0m
[1;35m[5m    Clave: 1: i[0m
    |---> SÃ­mbolo: i, CategorÃ­a: (13)VAR_FOR, Scope: 3, Type: mango, Informacion: 1 4 , Value: 1

[1;35m[5m    Clave: 2: mango[0m
    |---> SÃ­mbolo: mango, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 3: manguita[0m
    |---> SÃ­mbolo: manguita, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 4: manguangua[0m
    |---> SÃ­mbolo: manguangua, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 5: n[0m
    |---> SÃ­mbolo: n, CategorÃ­a: (1)VARIABLE, Scope: 2, Type: mango, Value: 6

[1;35m[5m    Clave: 6: negro[0m
    |---> SÃ­mbolo: negro, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 7: tas_claro[0m
    |---> SÃ­mbolo: tas_claro, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 8: hablame[0m
    |---> SÃ­mbolo: hablame, CategorÃ­a: (2)FUNCTION, Scope: 0

[1;35m[5m    Clave: 9: un_coÃ±o[0m
    |---> SÃ­mbolo: un_coÃ±o, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 10: array$[0m
    |---> SÃ­mbolo: array$, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 11: funcion$[0m
    |---> SÃ­mbolo: funcion$, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 12: error$[0m
    |---> SÃ­mbolo: error$, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 13: higuerote[0m
    |---> SÃ­mbolo: higuerote, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 14: rescata[0m
    |---> SÃ­mbolo: rescata, CategorÃ­a: (2)FUNCTION, Scope: 0

[1;35m[5m    Clave: 15: se_prende[0m
    |---> SÃ­mbolo: se_prende, CategorÃ­a: (2)FUNCTION, Scope: 0

--> Print AST? (s/n): [1;36m[5m
               =======================================================
                                   AST Representation                 
               =======================================================
[0m
Programa
|--> Main
     |--> InstrucciÃ³n | Kind: ;
          |--> AsignaciÃ³n | Kind: =
          |    |--> n | Category: DeclaraciÃ³n | Type: mango | Kind: VARIABLE
          |    |--> Literal | Category: NumÃ©rico | Type: mango | Value: 5
          |--> Bucle | Category: Determinado
               |--> i | Category: Var_Ciclo | Type: mango | Value: 1
               |    |--> entre | Category: Rango
               |    |    |--> Literal | Category: NumÃ©rico | Type: mango | Value: 1
               |    |--> hasta | Category: Rango
               |         |--> Literal | Category: NumÃ©rico | Type: mango | Value: 4
               |--> AsignaciÃ³n | Type: mango | Kind: = | Value: 6
                    |--> n | Category: Identificador | Type: mango | Value: 0
                    |--> + | Category: OperaciÃ³n | Type: mango | Kind: NumÃ©rica | Value: 6
                         |--> n | Category: Identificador | Type: mango | Value: 5
                         |--> i | Category: Identificador | Type: mango | Value: 1
[1;32m[5m
               =======================================================
                       ---->       Correct Program       <----        
               =======================================================
[0m
| Hay un poquito de cÃ³digo en tu marihuana d:-]
